lib/src/gotrue_client.dart

```diff
@@ -464,7 +468,12 @@ class GoTrueClient {
     }
 
     final response = await api.refreshAccessToken(token, jwt);
-    if (response.error != null) return response;
+    if (response.error != null) {
+      if (response.error!.statusCode == 'SocketException') {
+        _setTokenRefreshTimer(const Duration(seconds: 5));
```

Comment:
Here I have set it so that it will retry to get new access token every 5 seconds, but not sure if this is a good number. Open to suggestions.